### DISCUSSION TITLE
January CP2AIR FW

### DISCUSSION THREAD

**Discussion by simon0117** [ludwig-v/wireless-carplay-dongle-reverse-engineering#160] [General]
A week or so ago I used the OEM update on my Carplay2Air dongle, but my iPhone ended up being unable to connect wirelessly (I could only get wired passthrough CarPlay to work). I found this repo with the Firmware dumps, and decided to go straight to the most recent firmware with what I have assumed is U2W_CP2AIR_Update_2022.01.24.1914.img . While theUSB update was successful, when I checked the OEM update again it was a completely different interface than I had seen before, except all the detail information was missing and the update check did not work, the only thing that "worked" was the reset button which of course I tapped and had to re-add my iPhone through the car interface (2019 Toyota RAV4 hybrid) which was no big deal... except it seems like it will only work wirelessly, when I plug in the USB passthrough if the car is turned on, it never connects so I spend several minutes unplugging and replugging after connecting wirelessly if I want to get it connected wired. Phew! So I'm thinking maybe this 2022.01 version is not the best and I will be attempting a downgrade to U2W_Update_2021.08.30.0022.img (after the warning says downgrade to U2W_Update_2020.11.10.2138_ROLLBACK.img first?)

Maybe that's the problem, to correctly update to U2W_CP2AIR_Update_2022.01.24.1914.img I need to flash U2W_Update_2020.11.10.2138_ROLLBACK.img first?

![IMG_4445](https://user-images.githubusercontent.com/5805263/192051086-27d6c410-83d6-4ddc-ac19-f8e87f355356.PNG)

![IMG_4446](https://user-images.githubusercontent.com/5805263/192051105-0f96a336-85c6-4386-855e-4cd5e2e3dc4d.PNG)

![IMG_4448](https://user-images.githubusercontent.com/5805263/192051110-ef1d5b28-881a-4a65-b308-422100357b71.PNG)

![IMG_4450](https://user-images.githubusercontent.com/5805263/192051118-5705d916-a1af-4118-802e-c7af44f7fd07.PNG)

**Comment by simon0117** [comment]
Quick update, I flashed the rollback image and it worked, then I went through several upgrades to get to U2W_Update_2021.08.30.0022.img (I presume), but then it said there was another upgrade which downloaded but failed.

> Discover a new version
> VER:2021.12.22.1643 | Size:0.44MB
> Release Notes:
>  		Optimize memory usage
> 

I tried with the dongle connected to my computer (Mac) as well as the normal way in the car.

In the car, I seem to have the same problem as before where starting the car with the phone wired into the dongle's USB says that it's connecting but never does. This may be because I'm testing this out in the garage while still connected to my home wifi, I dunno. During the upgrades I noticed the wifi SSID was sometimes AutoBox-582f, sometimes TOYOTA86 which I think is the "correct" one. Playing around with the wifi on the phone between my home wifi and the dongle wifi I get it to connect wirelessly quicker, but most of the time when I plugged it into the USB it would disconnect then reconnect wireless, indicating it was just using the USB for charging.

I'm not sure how to troubleshoot this exactly so I can see exactly what is going on with the failed upgrade or the wired/wireless connection. I could do stuff on my Mac like monitor the HTTP requests or something

![IMG_4455](https://user-images.githubusercontent.com/5805263/192060264-d0791e83-8999-40cd-894f-8f34def4509b.PNG)

**Comment by ste-sanjaysingh** [comment]
+ Following : Experiencing same issue with Toyota and Cplay2air 2018 dongle.
